src/plugins/data/public/ui/dataset_selector/configurator.tsx

```diff
@@ -42,8 +47,20 @@ export const Configurator = ({
   const indexPatternsService = getIndexPatterns();
   const type = queryString.getDatasetService().getType(baseDataset.type);
   const languages = type?.supportedLanguages(baseDataset) || [];
+  const [selectIndexedView, setSelectIndexedView] = useState(false);
 
-  const [dataset, setDataset] = useState<Dataset>(baseDataset);
+  const [language, setLanguage] = useState<string>(() => {
+    const currentLanguage = queryString.getQuery().language;
+    if (languages.includes(currentLanguage)) {
+      return currentLanguage;
+    }
+    return languages[0];
+  });
+
+  const [dataset, setDataset] = useState<Dataset>({
+    ...cloneDeep(baseDataset),
+    language,
+  });
```

Comment:
avoid clone on re-render?
```suggestion
  const [dataset, setDataset] = useState<Dataset>(() => ({
    ...cloneDeep(baseDataset),
    language,
  }));
```

Reply:
The state won't get initialized again on re-render even if it is an object